src/EnergyPlus/GeneratorDynamicsManager.cc

```diff
@@ -137,39 +108,16 @@ namespace GeneratorDynamicsManager {
         // like a get input routine but feeds from
         //  parent objects, could have its own input object someday
 
-        // REFERENCES:
-        // na
-
-        // USE STATEMENTS:
-        // na
-
-        // Locals
-        // SUBROUTINE ARGUMENT DEFINITIONS:
-
-        // SUBROUTINE PARAMETER DEFINITIONS:
-        // na
-
-        // INTERFACE BLOCK SPECIFICATIONS:
-        // na
-
-        // DERIVED TYPE DEFINITIONS:
-        // na
-
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
 
         // get the number of generators that might use this module
         NumGensWDynamics = MicroCHPElectricGenerator::NumMicroCHPs; // TODO  + NumFuelCellCGenerators
 
         if (!allocated(GeneratorDynamics)) {
             GeneratorDynamics.allocate(NumGensWDynamics);
-        } else {
-            GeneratorDynamics.deallocate();
-            GeneratorDynamics.allocate(NumGensWDynamics);
```

Comment:
This is the fix. This function is called multiple times, so it should only allocate `GeneratorDynamics` once. These lines were clearing the data for prior generators, so only the last generator would be populated.